db/migrate/20221122015259_remove_archive_is_annotation_type.rb

```diff
@@ -0,0 +1,9 @@
+class RemoveArchiveIsAnnotationType < ActiveRecord::Migration[5.2]
+  class DynamicAnnotationType < ActiveRecord::Base
+    self.table_name = :dynamic_annotation_annotation_types
+  end
```

Comment:
Is this needed if you use `DynamicAnnotation::AnnotationType` below, in the `change` method?

Reply:
@caiosba I prefer not to use application models in migrations, since model names and methods can change over time and break past migrations. I think using a temporary class is considered a migration best practice, since it ensures that we can always run the migration in the future even if the model changes significantly.

here's a blogpost that talks about the approach: https://jakeyesbeck.com/2021/04/10/avoid-models-in-migrations/

another option I've used with the same effect is doing data migrations as idempotent, automatically run on deploy rake tasks that are periodically deleted.

Reply:
Ah right, that makes sense and I agree it's a good practice!

When we have long-running tasks like that, we don't implement as migrations, we implement as rake tasks that are executed after a deployment: https://github.com/meedan/check-api/tree/develop/lib/tasks/migrate. But since this one is super fast, I think it's better to keep it as a migration but using your temporary class approach.

Reply:
okay great. I didn't realize we also had that practice, good to know!